cmake/LibeventConfig.cmake.in

```diff
@@ -1,18 +1,178 @@
 # - Config file for the Libevent package
 # It defines the following variables
-#  LIBEVENT_INCLUDE_DIRS     - include directories
+#  LIBEVENT_FOUND            - true if libevent and all required components found on the system
+#  LIBEVENT_xxx_FOUND        - true if component xxx(see available components) found on the system
+#  LIBEVENT_VERSION          - libevent version in format Major.Minor.Patch
+#  LIBEVENT_INCLUDE_DIRS     - directories where libevent header is located.
+#  LIBEVENT_INCLUDE_DIR      - same as DIRS
+#  LIBEVENT_LIBRARIES        - libevent library to link against.
+#  LIBEVENT_LIBRARY          - same as LIBRARIES
+#
+# These variables are deprecated, only reserved for compatibility.
 #  LIBEVENT_STATIC_LIBRARIES - libraries to link against (archive/static)
 #  LIBEVENT_SHARED_LIBRARIES - libraries to link against (shared)
+#
+# When you try to locate the libevent libraries, you should specify which components you want to use.
+# The following table lists all available components. If none is given, all imported targets will used.
+#  event_core        - the core functons of libevent
+#  event_extra       - extra functions, contains http, dns ans rpc
+#  event             - sum of two above, deprecated
+#  event_pthreads    - multiple threads for libevent, not exists on Windows
+#  event_openssl     - openssl support for libevent
+#
+# By default, the shared libraries of libevent will be found. To find the static ones instead,
+# you must set the LIBEVENT_STATIC_LINK variable to TRUE before calling find_package(libevent ...).
+# example:
+#  set(LIBEVENT_STATIC_LINK TRUE)
+#  find_package(libevent 2.2 REQUIRED COMPONENTS event_core event_extra)
```

Comment:
Just a note, it will be nice to export `PACKAGE_VERSION` with the number of commits since the last tag, to make the following possible:

```
find_package(libevent 2.2.xx REQUIRED) 
```

(But should be a separate patch)

Reply:
I'll look at this.